backword-compatibility/class-adst-plugins-stats-api-bcp.php

```diff
@@ -0,0 +1,631 @@
+		add_shortcode( 'adv_stats_ratings', array( $this, 'display_plugin__ratings' ) );
+		add_shortcode( 'adv_stats_ratings_5star', array( $this, 'display_plugin__five_star_ratings' ) );
+		add_shortcode( 'adv_stats_ratings_average', array( $this, 'display_plugin__average_ratings' ) );
+		add_shortcode( 'adv_stats_plugin_ratings_average_in_star', array( $this, 'display_plugin_average_ratings_in_star' ) );
+		add_shortcode( 'adv_stats_downloads', array( $this, 'display_plugin__totaldownloads' ) );
+		add_shortcode( 'adv_stats_last_updated', array( $this, 'display_plugin__lastupdated' ) );
+		add_shortcode( 'adv_stats_download_link', array( $this, 'display_plugin__downloadlink' ) );
+		add_shortcode( 'adv_stats_downloads_counts', array( $this, 'display_download_count' ) );
+		add_shortcode( 'adv_stats_total_active', array( $this, 'display_active_installs' ) );
+	}
+
+
+	/**
+	 * Get the plugin Details.
+	 *
+	 * @param int $action Get attributes plugin Details.
+	 * @param int $api_params Get attributes plugin Details.
+	 * @return array $plugin Get plugin Details.
+	 */
+	public function get_api_data( $action, $api_params = array() ) {
+		$plugin_slug = isset( $api_params['plugin'] ) ? $api_params['plugin'] : '';
+
+		$second = 0;
+
+		$day = 0;
+
+		$adst_frequency = get_option( 'adst_info' );
+
+		if ( ! empty( $adst_frequency['Frequency'] ) ) {
+			$day = ( ( $adst_frequency['Frequency'] * 24 ) * 60 ) * 60;
+
+			$second = ( $second + $day );
+		}
+
+		// Get the plugin data if it has already been stored as a transient.
+		$plugin_data = get_transient( 'new_bsf_tr_plugin_info_' . esc_attr( $plugin_slug ) );
+
+		// If there is no transient, get the plugin data from wp.org.
+		if ( false === $plugin_data ) {
+			$response = wp_remote_get( 'http://api.wordpress.org/plugins/info/1.0/' . esc_attr( $plugin_slug ) . '.json?fields=active_installs' );
+
+			$plugin_data = (array) json_decode( wp_remote_retrieve_body( $response ) );
+
+			// If error in response or plugin slug incorrectly.
+			if ( is_wp_error( $response ) || empty( $plugin_data ) ) {
+				return __( 'Plugin slug is incorrect!', 'wp-themes-plugins-stats' );
+			}
+
+			$slug          = 'bsf_tr_plugin_info_' . $plugin_slug;
+			$update_option = array(
+				'slug'   => ( ! empty( $slug ) ? sanitize_text_field( $slug ) : '' ),
+				'plugin' => ( ! empty( $plugin_data ) ? $plugin_data : '' ),
+			);
+			update_option( 'adst_plugin_info', $update_option );
+
+			$plugin_db_data = get_option( 'adst_plugin_info' );// DB value.
+
+			$second = ( ! empty( $second ) ? $second : 86400 );
+			set_transient( 'new_bsf_tr_plugin_info_' . esc_attr( $plugin_slug ), $plugin_data, $second );
+		} else {
+			$second = ( ! empty( $second ) ? $second : 86400 );
+			set_transient( 'new_bsf_tr_plugin_info_' . esc_attr( $plugin_slug ), $plugin_data, $second );
+			$plugin_data = get_transient( 'new_bsf_tr_plugin_info_' . esc_attr( $plugin_slug ) );
```

Comment:
You have already fetched this transient data above here - https://github.com/brainstormforce/wp-themes-plugins-stats/pull/8/files#diff-b604deb35896b0ef9f33dc8bc66cbbf2R86